src/python/pants/util/ordered_set.py

```diff
@@ -206,6 +208,10 @@ def symmetric_difference_update(self, other: Iterable[T]) -> None:
         for item in items_to_add:
             self._items[item] = None
 
+    def sort(self, *, key: Optional[Callable] = None, reverse: bool = False):
+        """This method sorts the items in place in this set."""
+        self._items = {item: None for item in sorted(self._items, key=key, reverse=reverse)}  # type: ignore[type-var, arg-type]
```

Comment:
I didn't have the brain bandwidth right now to figure out how to satisfy MyPy here... 
```
src/python/pants/util/ordered_set.py: note: In member "sort" of class "OrderedSet":
src/python/pants/util/ordered_set.py:213:47: error: Value of type variable
"SupportsLessThanT" of "sorted" cannot be "T"  [type-var]
            self._items = {item: None for item in sorted(self._items, key=...
                                                  ^
src/python/pants/util/ordered_set.py:213:71: error: Argument "key" to "sorted"
has incompatible type "Optional[Callable[..., Any]]"; expected "None" 
[arg-type]
    ... {item: None for item in sorted(self._items, key=key, reverse=reverse)...
```